core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -208,7 +211,7 @@ private[spark] class TaskSchedulerImpl(
         taskSetsByStageIdAndAttempt.getOrElseUpdate(stage, new HashMap[Int, TaskSetManager])
       stageTaskSets(taskSet.stageAttemptId) = manager
       val conflictingTaskSet = stageTaskSets.exists { case (_, ts) =>
-        ts.taskSet != taskSet && !ts.isZombie
+        ts.taskSet != manager.taskSet && !ts.isZombie
```

Comment:
why this change?  `taskSet` should be the same as `manager.taskSet`, as `manager` is created from `taskSet` with `val manager = createTaskSetManager(taskSet, maxTaskFailures)`

Reply:
I have the same question, why use `manager.taskSet` here?

Reply:
Previously, I filter the already completed tasks in `createTaskSetManager()` and creates a new TaskSet by filtered tasks. So, the submitted TaskSet form DAGScheduler is not the same with TaskSet in TaskSetManager. But it failed plenty of unit tests since those tests would keep the original submitted TaskSet for some operations.
And now I have moved the `filter` logic into the TaskSetManager.